contracts/src/v0.8/keystone/CapabilityRegistry.sol

```diff
@@ -706,9 +723,23 @@ contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
   ) internal {
     DONCapabilityConfig storage donCapabilityConfig = s_dons[donId].config[configCount];
 
+    // Skip removing supported DON Ids from previously configured nodes in DON if
+    // we are adding the DON for the first time
+    if (configCount > 1) {
+      DONCapabilityConfig storage prevDONCapabilityConfig = s_dons[donId].config[configCount - 1];
+
+      // We acknowledge that this may result in an out of gas error if the number of configured
+      // nodes is large.  This is mitigated by ensuring that there will not be a large number
+      // of nodes configured to a DON
+      for (uint256 i; i < prevDONCapabilityConfig.nodes.length(); ++i) {
+        s_nodes[prevDONCapabilityConfig.nodes.at(i)].supportedDONIds.remove(donId);
```

Comment:
I expect that in most cases the new set of nodes will be the same as the old set of nodes (only some config fields will change). Can we do anything to optimize for that case and avoid removal+addition of every node?

Reply:
I guess we could encode and then hash the nodes array but that would require the order of node IDs to be the same